### ISSUE TITLE
There should be way to access app (vue prototype) in beforeRouteEnter

### ISSUE THREAD

**Issue by tjk** [vuejs/vue-router-next#1068]
### What problem does this feature solve?
During SSR, we use `beforeRouteEnter` so we do not begin rendering before finding out whether we might want to show a 404 page. All the "globals" are either provided or attached to globalProperties, and we need a way to access them.

### What does the proposed API look like?
Need to think about this more / get some feedback on the ask first.

**Comment by tjk** [comment]
For now, my workaround looks like this --

```
// app.js
const vuePrototype = app.config.globalProperties
router.beforeEach(to => {
  to.meta.vuePrototype = vuePrototype
})

// component.js
beforeRouteEnter(to, from, next) {
  // use to.meta.vuePrototype
},
```

**Comment by posva** [comment]
Duplicate of https://github.com/vuejs/vue-router-next/issues/700

**Comment by tjk** [comment]
@posva what is the point of just marking this as a duplicate and closing when the other issue is closed as well?

**Comment by posva** [comment]
Just read it on the other issue? 😕

**Comment by tjk** [comment]
The other issue says to attach the `app` instance to `router`. 

Component `beforeRouteEnter` does not even have access to `router` so I'm not sure how that is relevant for this issue.
